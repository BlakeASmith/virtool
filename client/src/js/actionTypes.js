/**
 * Created by igboyes on 03/05/17.
 */

const createRequestActionType = (root) => {
    return {
        REQUESTED: `${root}_REQUESTED`,
        SUCCEEDED: `${root}_SUCCEEDED`,
        FAILED: `${root}_FAILED`
    };
};

// Virus actionTypes
export const WS_UPDATE_VIRUS = "WS_UPDATE_VIRUS";
export const WS_REMOVE_VIRUS = "WS_REMOVE_VIRUS";

export const FIND_VIRUSES = createRequestActionType("FIND_VIRUSES");

export const GET_VIRUS_REQUESTED = "GET_VIRUS_REQUESTED";
export const GET_VIRUS_SUCCEEDED = "GET_VIRUS_SUCCEEDED";
export const GET_VIRUS_FAILED = "GET_VIRUS_FAILED";

export const CREATE_VIRUS_SET_NAME = "CREATE_VIRUS_SET_NAME";
export const CREATE_VIRUS_SET_ABBREVIATION = "CREATE_VIRUS_SET_ABBREVIATION";
export const CREATE_VIRUS_CLEAR = "CREATE_VIRUS_CLEAR";
export const CREATE_VIRUS_REQUESTED = "CREATE_VIRUS_REQUESTED";
export const CREATE_VIRUS_SUCCEEDED = "CREATE_VIRUS_SUCCEEDED";
export const CREATE_VIRUS_FAILED = "CREATE_VIRUS_FAILED";

export const EDIT_VIRUS_REQUESTED = "EDIT_VIRUS_REQUESTED";
export const EDIT_VIRUS_SUCCEEDED = "EDIT_VIRUS_SUCCEEDED";
export const EDIT_VIRUS_FAILED = "EDIT_VIRUS_FAILED";

export const REMOVE_VIRUS_REQUESTED = "REMOVE_VIRUS_REQUESTED";
export const REMOVE_VIRUS_SUCCEEDED = "REMOVE_VIRUS_SUCCEEDED";
export const REMOVE_VIRUS_FAILED = "REMOVE_VIRUS_FAILED";

// History actionTypes
export const FIND_HISTORY_REQUESTED = "FIND_HISTORY_REQUESTED";
export const FIND_HISTORY_SUCCEEDED = "FIND_HISTORY_SUCCEEDED";
export const FIND_HISTORY_FAILED = "FIND_HISTORY";

export const REMOVE_CHANGE_REQUESTED = "REMOVE_CHANGE_REQUESTED";
export const REMOVE_CHANGE_SUCCEEDED = "REMOVE_CHANGE_SUCCEEDED";
export const REMOVE_CHANGE_FAILED = "REMOVE_CHANGE_FAILED";

// Users actionTypes
export const LOAD_USERS = "LOAD_USERS";
export const CREATE_USER = "CREATE_USER";

// Account actionTypes
export const GET_ACCOUNT_REQUESTED = "GET_ACCOUNT_REQUESTED";
export const GET_ACCOUNT_SUCCEEDED = "GET_ACCOUNT_SUCCEEDED";
export const GET_ACCOUNT_FAILED = "GET_ACCOUNT_FAILED";
export const LOGOUT_REQUESTED = "LOGOUT_REQUESTED";
export const LOGOUT_SUCCEEDED = "LOGOUT_SUCCEEDED";

// Administrative setting actionTypes
export const GET_SETTINGS = createRequestActionType("GET_SETTINGS");
export const UPDATE_SETTINGS = createRequestActionType("UPDATE_SETTINGS");

export const SET_SOURCE_TYPE_VALUE = "SET_SOURCE_TYPE_VALUE";
export const SET_CONTROL_READAHEAD_TERM = "SET_CONTROL_READAHEAD_TERM";
export const GET_CONTROL_READAHEAD_REQUESTED = "GET_CONTROL_READAHEAD_REQUESTED";
export const GET_CONTROL_READAHEAD_SUCCEEDED = "GET_CONTROL_READAHEAD_SUCCEEDED";
export const GET_CONTROL_READAHEAD_FAILED = "GET_CONTROL_READAHEAD_FAILED";

export const LIST_USERS = createRequestActionType("LIST_USERS");
export const SELECT_USER = createRequestActionType("SELECT_USER");
export const CHANGE_SET_PASSWORD = "CHANGE_SET_PASSWORD";
export const CLEAR_SET_PASSWORD = "CLEAR_SET_PASSWORD";
export const SET_FORCE_RESET = createRequestActionType("SET_FORCE_RESET");
