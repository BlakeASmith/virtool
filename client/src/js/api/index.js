/**
 *
 *
 * @copyright 2017 Government of Canada
 * @license MIT
 * @author igboyes
 *
 */

export * from "./viruses";
