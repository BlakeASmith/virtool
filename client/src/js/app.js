/**
 * Created by igboyes on 03/03/15.
 */

var React = require('react');
var ReactDOM = require('react-dom');
var Start = require('./components/Start.jsx');

document.write(require('../css/bootstrap.css'));
document.write(require('../css/font.css'));
document.write(require('../css/perfect-scrollbar.css'));
document.write(require('../css/typeahead.css'));
document.write(require('../css/graphics.css'));
document.write(require('../css/style.css'));

ReactDOM.render(React.createElement(Start), document.getElementById('app-container'));