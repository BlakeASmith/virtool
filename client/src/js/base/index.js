import { Alert } from "./Alert";
import { AlgorithmSelect } from "./AlgorithmSelect";
import { AutoProgressBar, ProgressBar } from "./ProgressBar";
import { bsStyles } from "./utils";
import { Button } from "./Button";
import { Checkbox } from "./Checkbox";
import { Flex, FlexItem } from "./Flex";
import { Help } from "./Help";
import { Icon } from "./Icon";
import { Identicon } from "./Identicon";
import { IDRow } from "./IDRow";
import { Input } from "./Input";
import { InputError } from "./InputError";
import { InputSave } from "./InputSave";
import { ListGroupItem } from "./ListGroupItem";
import { LoadingPlaceholder } from "./LoadingPlaceholder";
import { NoneFound } from "./NoneFound";
import { Pagination } from "./Pagination";
import { Radio } from "./Radio";
import { RelativeTime } from "./RelativeTime";
import { RemoveModal } from "./RemoveModal";
import { Tooltip } from "./Tooltip";
import { ViewHeader } from "./ViewHeader";

export {
    Alert,
    AlgorithmSelect,
    AutoProgressBar,
    bsStyles,
    Button,
    Checkbox,
    Flex,
    FlexItem,
    Help,
    Icon,
    Identicon,
    IDRow,
    Input,
    InputError,
    InputSave,
    ListGroupItem,
    LoadingPlaceholder,
    NoneFound,
    Pagination,
    ProgressBar,
    Radio,
    RelativeTime,
    RemoveModal,
    Tooltip,
    ViewHeader
};
