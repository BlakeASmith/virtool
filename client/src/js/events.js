/**
 * Copyright 2015, Government of Canada.
 * All rights reserved.
 *
 * This source code is licensed under the MIT license.
 *
 * @providesModule Events *
 */

import { transform, remove, each } from "lodash";

/**
 * A simple event managing object that can be bound into other objects to make them capable of managing callback and
 * emitting events.
 *
 * @param eventTypes {array} - An array of event names that can be bound and emitted by the Events object.
 * @param parent {object} - A parent object to forward the on, off, and emit functions to.
 * @class
 */
class Events {

    constructor (eventTypes, parent) {

        // Add a property to this.bound for each event type. The key is the eventType and the value is a list that will
        // all bound callbacks for that eventType.
        this.bound = transform(eventTypes, (result, eventType) => result[eventType] = [], {});

        // Bind the following function to a passed parent object to make it directly able to emit events and bind
        // callbacks. Bind the methods to the Events object.
        if (parent) {
            parent.on = this.on.bind(this);
            parent.off = this.off.bind(this);
            parent.emit = this.emit.bind(this);
        }
    }

    /**
     * Bind a callback to an event.
     *
     * @param event {string} - The event to bind the callback to.
     * @param callback {function} - The function to call when the event is emitted.
     * @func
     */
    on (event, callback) {
        if (this.bound.hasOwnProperty(event)) {
            this.bound[event].push(callback);
        } else {
            throw new Error(`Attempted to bind callback to non-existent event "${event}"`);
        }
    }

    /**
     * Unbind a callback from an event.
     *
     * @param event {string} - The event the callback should be unbound from.
     * @param callback {function} - The callback that should be unbound.
     * @func
     */
    off (event, callback) {
        remove(this.bound[event], (candidateCallback) => callback === candidateCallback);
    }

    /**
     * Emits and event. This entails calling every callback function that is bound to the event. Optional data can be
     * passed as an argument to the callback.
     *
     * @param events {array} the name of the event to emit.
     * @param data {object} a data object that can be passed to the bound callbacks when they are called.
     * @func
     */
    emit (events, data) {
        const eventsArray = events instanceof Array ? events: [events];
        eventsArray.forEach((event) => each(this.bound[event], (callback) => callback(data)));
    }
}

export default Events;
