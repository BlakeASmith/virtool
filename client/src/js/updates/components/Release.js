/**
 *
 *
 * @copyright 2017 Government of Canada
 * @license MIT
 * @author igboyes
 *
 */

import React, { PropTypes } from "react";
import Marked from "marked";
import { ListGroupItem, Button, Row, Col } from "react-bootstrap";

const renderReleaseMarkdown = (body) => {
    let html = Marked(body);

    html = html.replace(/#([0-9]+)/g, `<a target="_blank" href="https://github.com/virtool/virtool/issues/$1">#$1</a>`);

    return <div style={{marginTop: "10px"}} dangerouslySetInnerHTML={{__html: html}} />;
};

export default class Release extends React.Component {

    constructor (props) {
        super(props);

        this.state = {
            in: false
        };
    }

    static propTypes = {
        name: PropTypes.string,
        body: PropTypes.string
    };

    render () {
        return (
            <ListGroupItem>
                <Row>
                    <Col xs={12}>
                        <strong>{this.props.name}</strong>
                        <span className="pull-right">
                            <Button bsSize="xsmall" target="_blank" href={this.props.html_url}>
                                <i className="i-github" /> GitHub
                            </Button>
                        </span>
                    </Col>
                    <Col xs={12}>
                        {renderReleaseMarkdown(this.props.body)}
                    </Col>
                </Row>
            </ListGroupItem>
        );
    }
}
