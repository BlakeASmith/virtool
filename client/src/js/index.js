import React from "react";
import ReactDOM from "react-dom";
import Raven from "raven-js";

import App from "./App";
import WSConnection from "./websocket";
import createHistory from "history/createBrowserHistory";
import { getAccount } from "./account/actions";
import { getSettings } from "./settings/actions";
import { assign } from "lodash";
import { createStore, combineReducers, applyMiddleware } from "redux";
import createSagaMiddleware from "redux-saga";
import { routerReducer, routerMiddleware } from "react-router-redux";

import { SET_APP_PENDING, UNSET_APP_PENDING } from "./actionTypes";
import jobsReducer from "./jobs/reducers";
import samplesReducer from "./samples/reducers";
import indexesReducer from "./indexes/reducers";
import virusesReducer from "./viruses/reducers";
import subtractionReducer from "./subtraction/reducers";
import filesReducer from "./files/reducer";
import accountReducer from "./account/reducers";
import settingsReducer from "./settings/reducers";
import usersReducer from "./users/reducers";
import groupsReducer from "./groups/reducers";
import updatesReducer from "./updates/reducers";
import rootSaga from "./sagas";

export * from "../style/style.less";

Raven.config("https://9a2f8d1a3f7a431e873207a70ef3d44d@sentry.io/220532").install();

window.Raven = Raven;

const sagaMiddleware = createSagaMiddleware();

const appInitialState = {
    pending: false
};

const appReducer = (state = appInitialState, action) => {

    switch (action.type) {

        case SET_APP_PENDING:
            return assign({}, state, {pending: true});

        case UNSET_APP_PENDING:
            return assign({}, state, {pending: false});
    }

    return state;
};

const history = createHistory();

const store = createStore(
    combineReducers({
        app: appReducer,
        jobs: jobsReducer,
        samples: samplesReducer,
        viruses: virusesReducer,
        indexes: indexesReducer,
        subtraction: subtractionReducer,
        files: filesReducer,
        settings: settingsReducer,
        users: usersReducer,
        groups: groupsReducer,
        account: accountReducer,
        updates: updatesReducer,
        router: routerReducer
    }),
    applyMiddleware(sagaMiddleware, routerMiddleware(history)),
);

sagaMiddleware.run(rootSaga);

window.store = store;

window.ws = new WSConnection(store.dispatch);
window.ws.establishConnection();

window.store.dispatch(getAccount());
window.store.dispatch(getSettings());

window.addEventListener("beforeunload", () => {
    if (!window.store.getState().files.uploadsComplete) {
        return "hello world";
    }
});

ReactDOM.render(
    <App store={store} history={history} />,
    document.getElementById("app-container")
);
