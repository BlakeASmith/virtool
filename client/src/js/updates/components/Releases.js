import { push } from "connected-react-router";
import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { Icon, Box } from "../../base";
import Install from "./Install";
import ReleasesList from "./List";

const StyledReleasesPanelBody = styled(Box)`
    padding: 10px 15px;
`;

export const Releases = ({ onShowInstall, releases }) => {
    if (releases.length) {
        return (
            <StyledReleasesPanelBody>
                <ReleasesList releases={releases} onShowInstall={onShowInstall} />
                <Install />
            </StyledReleasesPanelBody>
        );
    }

    return (
        <StyledReleasesPanelBody>
            <Icon bsStyle="success" name="check" />
            <strong className="text-success"> Software is up-to-date</strong>
        </StyledReleasesPanelBody>
    );
};

export const mapStateToProps = state => ({
    releases: state.updates.releases
});

export const mapDispatchToProps = dispatch => ({
    onShowInstall: () => {
        dispatch(push({ state: { install: true } }));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(Releases);
