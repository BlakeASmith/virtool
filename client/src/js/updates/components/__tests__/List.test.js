import React from "react";
import ReleasesList from "../List.js";
import Release from "../Release.js";
import { Releases } from "../Releases.js";

describe("<ReleasesList />", () => {
    let props;

    beforeEach(() => {
        props = {
            releases: [
                {
                    name: "foo",
                    body: "Foo",
                    html_url: "FOO"
                }
            ],
            onShowInstall: jest.fn()
        };
    });

    it("should render with one release", () => {
        const wrapper = shallow(<ReleasesList {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render with two releases", () => {
        props = {
            releases: [
                {
                    name: "foo",
                    body: "Foo",
                    html_url: "FOO"
                },
                {
                    name: "bar",
                    body: "Bar",
                    html_url: "BAR"
                }
            ],
            onShowInstall: jest.fn()
        };

        const wrapper = shallow(<ReleasesList {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should call onShowInstall when Button clicked", () => {
        const wrapper = shallow(<ReleasesList {...props} />);

        expect(props.onShowInstall).not.toHaveBeenCalled();
        wrapper.find("Button").simulate("click");
        expect(props.onShowInstall).toHaveBeenCalled();
    });
});
