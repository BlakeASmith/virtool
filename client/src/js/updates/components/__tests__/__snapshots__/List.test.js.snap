// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReleasesList /> should render with one release 1`] = `
<Box__BoxGroup>
  <List__ReleasesListHeader>
    <strong
      className="text-warning"
    >
      <Icon
        faStyle="fas"
        fixedWidth={false}
        name="arrow-alt-circle-up"
      />
       Update
       Available
    </strong>
    <Button
      color="blue"
      disabled={false}
      icon="download"
      onClick={[MockFunction]}
      tipPlacement="top"
      type="button"
    >
      Install
    </Button>
  </List__ReleasesListHeader>
  <Release
    body="Foo"
    html_url="FOO"
    key="foo"
    name="foo"
  />
</Box__BoxGroup>
`;

exports[`<ReleasesList /> should render with two releases 1`] = `
<Box__BoxGroup>
  <List__ReleasesListHeader>
    <strong
      className="text-warning"
    >
      <Icon
        faStyle="fas"
        fixedWidth={false}
        name="arrow-alt-circle-up"
      />
       Update
      s
       Available
    </strong>
    <Button
      color="blue"
      disabled={false}
      icon="download"
      onClick={[MockFunction]}
      tipPlacement="top"
      type="button"
    >
      Install
    </Button>
  </List__ReleasesListHeader>
  <Release
    body="Foo"
    html_url="FOO"
    key="foo"
    name="foo"
  />
  <Release
    body="Bar"
    html_url="BAR"
    key="bar"
    name="bar"
  />
</Box__BoxGroup>
`;
